assets/src/models/createDetourMachine.ts

```diff
@@ -1,28 +1,134 @@
-import { setup } from "xstate"
+import { setup, assign } from "xstate"
+import { ShapePoint } from "../schedule"
 
 export const createDetourMachine = setup({
   types: {} as {
-    events: { type: "detour.edit.done" } | { type: "detour.edit.resume" }
+    context: {
+      waypoints: ShapePoint[]
+    }
+    events:
+      | { type: "detour.edit.clear-detour" }
+      | { type: "detour.edit.done" }
+      | { type: "detour.edit.place-connection-point"; location: ShapePoint }
```

Comment:
We've discussed before that "Connection Point" actually means something different (the last un-missed stop before the detour begins and the first un-missed stop after the detour ends). Should we take this opportunity to come up with a different name?

```suggestion
      | { type: "detour.edit.place-point-on-original-route"; location: ShapePoint }
```
☝️ a possibility, although honestly... we can do better 😅

Reply:
how about `detour.edit.place-detour-terminus-point`?

Reply:
I'm mainly thinking that we may have "ghost" connection points, that are on the original route, so I'm trying to get at the root user intention.

Reply:
Though `terminus` is not great :/ looking for something to generalize the idea of "start" and "end"

Reply:
Oh, huh.

Maybe I'm missing something, but isn't that a different question? Like, we're displaying the points in one spot, and computing them effectively from a different spot, but either way, the question is... what should they be called?